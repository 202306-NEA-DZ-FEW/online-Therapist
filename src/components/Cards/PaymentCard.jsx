import Image from "next/image";
import { useState } from "react";

export default function PaymentCard({
    number,
    name,
    expDate,
    CardType,
    onClick,
}) {
    const [selected, setSelected] = useState(false);

    const handleCardClick = () => {
        setSelected(!selected);
        if (onClick) {
            onClick();
        }
    };

    // Define background colors for Master and Visa cards
    const cardColors = {
        mastercard: "from-[#FF9100] to-[#F7353B]",
        visa: "from-[#007FC1] to-[#6AAFDC]",
    };

    const borderClasses = selected
        ? "border-8 border-Teal"
        : "border border-gray-300";
    return (
        <div
            onClick={handleCardClick}
            className={`${borderClasses} w-[270px] h-[190px] md:w-[280px] md:h-[190px] lg:w-[380px] lg:h-[230px] bg-gradient-to-tr ${cardColors[CardType] || cardColors.visa} rounded-2xl flex content-center items-center justify-center`}
        >
            <div className='p-8 w-full h-full'>
                <div className='relative w-full h-full'>
                    <Image
                        className='absolute w-[45px] md:w-[45px] lg:w-[60px]'
                        alt=''
                        src={`/${CardType}.svg`}
                        width={60}
                        height={24}
                    />
                    <Image
                        className='absolute right-0 bottom-0 top-0 my-auto w-[40px] md:w-[40px] lg:w-[50px]'
                        alt='chip'
                        src='/chip.svg'
                        width={50}
                        height={30}
                    />
                    <div className='flex flex-col w-full h-full justify-end gap-4'>
                        <p className='text-lg text-white md:text-lg lg:text-2xl'>
                            {number}{" "}
                        </p>
                        <div className='flex gap-12 md:gap-12 lg:gap-28'>
                            <p className='text-md text-white md:text-md lg:text-lg uppercase'>
                                {name}{" "}
                            </p>
                            <p className='text-md text-white md:text-md lg:text-lg uppercase'>
                                {expDate}{" "}
                            </p>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
}
