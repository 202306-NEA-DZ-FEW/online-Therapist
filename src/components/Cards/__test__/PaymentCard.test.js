import renderer from "react-test-renderer";

import PaymentCard from "../PaymentCard";

it("renders correctly", () => {
    const tree = renderer.create(<PaymentCard />).toJSON();
    expect(tree).toMatchSnapshot();
});
