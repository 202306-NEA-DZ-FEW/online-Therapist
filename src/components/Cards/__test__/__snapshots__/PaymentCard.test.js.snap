// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="border border-gray-300 w-[270px] h-[190px] md:w-[280px] md:h-[190px] lg:w-[370px] lg:h-[220px] bg-gradient-to-tr from-[#007FC1] to-[#6AAFDC] rounded-2xl flex content-center items-center justify-center"
  onClick={[Function]}
>
  <div
    className="p-8 w-full h-full"
  >
    <div
      className="relative w-full h-full"
    >
      <img
        alt=""
        className="absolute w-[45px] md:w-[45px] lg:w-[60px]"
        data-nimg="1"
        decoding="async"
        height={24}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/undefined.svg"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={60}
      />
      <img
        alt="chip"
        className="absolute right-0 bottom-0 top-0 my-auto w-[40px] md:w-[40px] lg:w-[50px]"
        data-nimg="1"
        decoding="async"
        height={30}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/chip.svg"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={50}
      />
      <div
        className="flex flex-col w-full h-full justify-end gap-4"
      >
        <p
          className="text-lg text-white md:text-lg lg:text-2xl"
        >
           
        </p>
        <div
          className="flex gap-12 md:gap-12 lg:gap-28"
        >
          <p
            className="text-md text-white md:text-md lg:text-lg uppercase"
          >
             
          </p>
          <p
            className="text-md text-white md:text-md lg:text-lg uppercase"
          >
             
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
